private/buf/bufformat/formatter.go

```diff
@@ -1861,31 +1917,41 @@ func (f *formatter) writeLineEnd(node ast.Node) {
 //	// multiple lines.
 //	message Foo {}
 func (f *formatter) writeMultilineComments(comments ast.Comments) {
-	var previousComment ast.Comment
+	f.writeMultilineCommentsMaybeCompact(comments, false)
+}
+
+func (f *formatter) writeMultilineCommentsMaybeCompact(comments ast.Comments, forceCompact bool) {
+	compact := forceCompact || isOpenBrace(f.previousNode)
 	for i := 0; i < comments.Len(); i++ {
 		comment := comments.Index(i)
-		if i > 0 {
-			var (
-				previousCommentIsCStyle = cStyleComment(previousComment)
-				precedingNewlineCount   = newlineCount(comment.LeadingWhitespace())
-			)
-			if previousCommentIsCStyle && precedingNewlineCount > 1 ||
-				!previousCommentIsCStyle && precedingNewlineCount > 0 {
-				// Newlines between blocks of comments should be preserved.
-				//
-				// For example,
-				//
-				//  // This is a license header
-				//  // spread across multiple lines.
-				//
-				//  // Package pet.v1 defines a PetStore API.
-				//  package pet.v1;
-				//
-				f.P()
+		if !compact && newlineCount(comment.LeadingWhitespace()) > 1 {
+			// Newlines between blocks of comments should be preserved.
+			//
+			// For example,
+			//
+			//  // This is a license header
+			//  // spread across multiple lines.
+			//
+			//  // Package pet.v1 defines a PetStore API.
+			//  package pet.v1;
+			//
+			f.P()
+		}
+		compact = false
+		txt := comment.RawText()
+		if strings.HasPrefix(txt, "/*") && newlineCount(txt) > 0 {
+			lines := strings.Split(txt, "\n")
+			for j, line := range lines {
+				line = strings.TrimSpace(line)
+				if j == 0 || line == "*/" {
+					f.P(line)
+				} else {
+					f.P("   ", line)
```

Comment:
Does this need to take into account indentation level?

Reply:
It does. `f.P(...)` prints the indentation, then the given args, then a newline. (If not args, indentation is skipped and it just prints empty line.)